synse/http.go

```diff
@@ -103,41 +112,43 @@ func (c *httpClient) Config() (*Config, error) {
 	return out, nil
 }
 
-// getUnversioned makes an unversioned request.
-func (c *httpClient) getUnversioned(path string, okResp interface{}) error {
-	errResp := new(Error)
-
-	_, err := c.client.R().
-		SetResult(okResp).
-		SetError(errResp).
-		Get(path)
+// getUnversioned performs a GET request against the Synse Server unversioned API.
+func (c *httpClient) getUnversioned(uri string, okScheme interface{}) error {
+	errScheme := new(Error)
+	client, err := c.setUnversioned()
 	if err != nil {
-		return errors.Wrap(err, "failed to make an unversioned request to synse server")
-	}
-
-	if *errResp != (Error{}) {
-		return errors.Errorf("got an error response from synse server: %+v", errResp)
+		return errors.Wrap(err, "failed to set an unversioned host")
 	}
 
-	return nil
+	_, err = client.R().SetResult(okScheme).SetError(errScheme).Get(uri)
+	return check(err, errScheme)
 }
 
-// getVersioned makes a versioned request.
-func (c *httpClient) getVersioned(path string, okResp interface{}) error {
-	err := c.cacheAPIVersion()
+// getVersioned performs a GET request against the Synse Server versioned API.
+func (c *httpClient) getVersioned(uri string, okScheme interface{}) error {
+	errScheme := new(Error)
+	client, err := c.setVersioned()
 	if err != nil {
-		return errors.Wrap(err, "failed to cache api version")
+		return errors.Wrap(err, "failed to set a versioned host")
 	}
 
-	// FIXME - will want to separate out the URL construction for scalability
-	// when it comes to adding POST method.
-	versionedPath := fmt.Sprintf("%s/%s", c.apiVersion, path)
-	err = c.getUnversioned(versionedPath, okResp)
+	_, err = client.R().SetResult(okScheme).SetError(errScheme).Get(uri)
+	return check(err, errScheme)
+}
+
+// setUnversioned returns a client that uses unversioned host URL.
+func (c *httpClient) setUnversioned() (*resty.Client, error) {
+	return c.client.SetHostURL(fmt.Sprintf("http://%s/synse/", c.options.Address)), nil
```

Comment:
If we are always returning `nil` for the error, I'm not sure we need to return an error variable at all.